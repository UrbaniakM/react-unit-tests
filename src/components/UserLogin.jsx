import * as React from "react";

export const UserLogin = ({ url }) => {
  const [responseStatus, setResponseStatus] = React.useState(null);

  const makeApiCall = React.useCallback(async () => {
    const response = await fetch(url);

    setResponseStatus(response.status);
  }, [url]);

  return (
    <div>
      Click on the button to log in to <span>{url}</span>
      <button onClick={makeApiCall}>Log me in</button>
      {responseStatus && <div>Response status is <span>{responseStatus}</span></div>}
    </div>
  )
}
