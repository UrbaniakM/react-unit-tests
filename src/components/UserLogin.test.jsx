import { UserLogin } from "./UserLogin";

describe("UserLogin", () => {

  beforeAll(() => {
    // Mock responseStatus state and makeApiCall callback here.
    // To be able to do this, you have to move both of them out of UserLogin component.
  });
  
  beforeEach(() => {
    // You will have to check how many times makeApiCall was called, so it's worth to clear mocks between tests.
  });

  it("renders successfully", () => {
    // test if user login renders successfully
    // create snapshot test

    expect(false).toBeTruthy();
  });

  it("on click sends login request to URL passed as prop", () => {
    // test if makeApiCall was being called on click event

    expect(false).toBeTruthy();
  })
});
