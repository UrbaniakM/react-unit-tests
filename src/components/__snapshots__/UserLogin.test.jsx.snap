// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLogin renders successfully 1`] = `
<div>
  <div>
    Click on the button to log in to 
    <span>
      https://something.abcd.efgghfsa
    </span>
    <button>
      Log me in
    </button>
    <div>
      Response status is 
      <span>
        200
      </span>
    </div>
  </div>
</div>
`;
